pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -696,6 +704,11 @@ func (rd *restoreDataProcessor) Next() (rowenc.EncDatumRow, *execinfrapb.Produce
 		if !ok {
 			// Done. Check if any phase exited early with an error.
 			err := rd.phaseGroup.Wait()
+			if rd.progressMade {
+				err = errors.Mark(err, retryableRestoreProcError)
```

Comment:
I mildly wonder if retryableRestoreProcError should be a more generic thing that lives in the jobserror package that would then be handling by the IsPermanentBulkJobError method.  But I'm OK with this as is.

Reply:
Yeah, I'd like merge as is, since there's a chance we backport this at some point.